app/src/main/java/com/sun_asterisk/moviedb_50/data/source/remote/fetchjson/ParseJsonToMovie.kt

```diff
@@ -0,0 +1,67 @@
+package com.sun_asterisk.moviedb_50.data.source.remote.fetchjson
+
+import com.sun_asterisk.moviedb_50.data.model.Genres
+import com.sun_asterisk.moviedb_50.data.model.Movie
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+
+class ParseJsonToMovie {
+
+    fun parseJsonToData(jsonObject: JSONObject): MutableList<Movie>? {
+        val movieList: MutableList<Movie> = mutableListOf()
+        try {
+            val jsonArray = jsonObject.getJSONArray(Movie.MovieEntry.MOVIE)
+            for (i in 0 until jsonArray.length()) {
+                val movieJson = jsonArray.getJSONObject(i)
+                val movie: Movie? = movieJson?.let { parseJsonToObject(it) }
+                movie?.let { movieList.add(it) }
+            }
+        } catch (e: JSONException) {
+            e.printStackTrace()
+        }
+        return movieList
+    }
+
+    private fun parseJsonToObject(movieJson: JSONObject): Movie? {
+        try {
+            val id = movieJson.optInt(Movie.MovieEntry.ID)
+            val title = movieJson.optString(Movie.MovieEntry.TITLE)
+            val overview = movieJson.optString(Movie.MovieEntry.OVERVIEW)
+            val posterPath = movieJson.optString(Movie.MovieEntry.POSTER_PATH)
+            val backdropPath = movieJson.optString(Movie.MovieEntry.BACKDROP_PATH)
+            val voteCount = movieJson.optInt(Movie.MovieEntry.VOTE_COUNT)
+            val voteAverage = movieJson.optDouble(Movie.MovieEntry.VOTE_AVERAGE)
+            val popularity = movieJson.optDouble(Movie.MovieEntry.POPULARITY)
+            val releaseDate = movieJson.optString(Movie.MovieEntry.RELEASE_DATE)
+            val genre = getGenresIdArray(movieJson.optJSONArray(Genres.GenresEntry.GENRES))
+
+            return Movie(
+                id,
+                title,
+                overview,
+                posterPath,
+                backdropPath,
+                voteCount,
+                voteAverage,
+                popularity,
+                releaseDate,
+                genre
+            )
```

Comment:
Thêm 1 constructor trong class Movie để gọn đoạn này hơn
```suggestion
        try {     
            return Movie(movieJson)
```

khi ấy trong class Movie sẽ có thêm constructor 
```Kotlin
constructor(movieJson: JSONObject) : this(
    id = movieJson.optInt(Movie.MovieEntry.ID)
    title = movieJson.optString(Movie.MovieEntry.TITLE)
    overview = movieJson.optString(Movie.MovieEntry.OVERVIEW)
    posterPath = movieJson.optString(Movie.MovieEntry.POSTER_PATH)
    backdropPath = movieJson.optString(Movie.MovieEntry.BACKDROP_PATH)
    voteCount = movieJson.optInt(Movie.MovieEntry.VOTE_COUNT)
    voteAverage = movieJson.optDouble(Movie.MovieEntry.VOTE_AVERAGE)
    popularity = movieJson.optDouble(Movie.MovieEntry.POPULARITY)
    releaseDate = movieJson.optString(Movie.MovieEntry.RELEASE_DATE)
    genre = getGenresIdArray(movieJson.optJSONArray(Genres.GenresEntry.GENRES))
)
```